### PR TITLE
Numerically stable logit pred

### PR DESCRIPTION
**Pull request by dr-david**
The get_logit_predictions function was computing $\text{logit}(\text{softmax}(x))$ which is not stable for values of $\text{softmax}(x) \approx 0 \text{ or } \approx 1$

I rewrote it as:

$$
\text{logit}(\text{softmax}(x)_i) = \log(\text{softmax}(x)_i) - \log(1 - \text{softmax}(x)_i)
$$

Expanding further:

$$
\text{logit}(\text{softmax}(x)\_i) = x\_i - \text{logSumExp}(x) - \left(\text{logSumExp}(x\_{-i}) - \text{logSumExp}(x)\right)
= x\_i - \text{logSumExp}(x\_{-i})
$$

### CODE REVIEW

**Review by pawel-czyz — APPROVED** [review]
Looks good to me!

File: src/covvfit/_quasimultinomial.py [hunk 1]
```diff
@@ -359,22 +359,58 @@ def get_softmax_predictions(
     y_softmax = jax.nn.softmax(y_linear, axis=-1)
     return y_softmax
 
+def _logsumexp_excluding_column(
```
[the remark this answers is not in the document]
Comment (pawel-czyz) [reply src/covvfit/_quasimultinomial.py:363]: Addressed :)
